expression/generator/time_vec.go

```diff
@@ -249,6 +249,184 @@ func (b *{{.SigName}}) vectorized() bool {
+			{{- template "BufAllocator1" . }}
+		{{- end }}
+		
+		if err := b.args[0].VecEval{{ .TypeA.TypeName }}(b.ctx, input, buf0); err != nil {
+			return err
+		}
+		if err := b.args[1].VecEval{{ .TypeB.TypeName }}(b.ctx, input, buf1); err != nil {
+			return err
+		}
+
+		{{- if and .TypeA.Fixed .TypeB.Fixed }}
+			result.MergeNulls(buf0, buf1)
+			arg0 := buf0.{{.TypeA.TypeNameInColumn}}s()
+			arg1 := buf1.{{.TypeB.TypeNameInColumn}}s()
+		{{- else }}
+			{{- if .TypeA.Fixed }} 
+				result.MergeNulls(buf0)
+				arg0 := buf0.{{.TypeA.TypeNameInColumn}}s()
+			{{- end }}
+			{{- if .TypeB.Fixed }} 
+				result.MergeNulls(buf1)
+				arg1 := buf1.{{.TypeB.TypeNameInColumn}}s()
+			{{- end }}
+		{{- end }}
+
+		{{- if (or $AIsDuration $BIsDuration) }} 
+			var (
+				lhs    types.Duration
+				rhs    types.Duration
+			)
+		{{- end }}
+	for i:=0; i<n ; i++{
+		{{- if and .TypeA.Fixed .TypeB.Fixed }} 
+			if result.IsNull(i) {
+		{{- else if .TypeA.Fixed }} 
+			if result.IsNull(i) || buf1.IsNull(i) {
+		{{- else if .TypeB.Fixed }} 
+			if result.IsNull(i) || buf0.IsNull(i) {
+		{{- else }} 
+			if buf1.IsNull(i) || buf0.IsNull(i) {
+			result.SetNull(i, true)
+		{{- end }}
+			continue
+		}
+		
+		{{- if $AIsString }}
+			{{ if $BIsDuration }} lhsDur, _, lhsIsDuration,
+			{{- else if $BIsTime }} _, lhsTime, lhsIsDuration,
+			{{- else if $BIsString }} lhsDur, lhsTime, lhsIsDuration,
+			{{- end }}  err := convertStringToDuration(b.ctx.GetSessionVars().StmtCtx, buf0.GetString(i), int8(b.tp.Decimal))
+			if err != nil  {
+				return err
+			}
+			{{- if $BIsDuration }}
+			if !lhsIsDuration {
+				result.SetNull(i, true)
+				continue
+			}
+			lhs = lhsDur
+			{{- else if $BIsTime }}
+			if lhsIsDuration {
+				result.SetNull(i, true)
+				continue
+			}
+			{{- end }}
+		{{- else if $AIsTime }} 
+			lhsTime := arg0[i]
+		{{- else }} 
+			lhs.Duration = arg0[i]
+		{{- end }}
+
+		{{- if $BIsString }}
+			{{ if $AIsDuration }} rhsDur, _, rhsIsDuration,
+			{{- else if $AIsTime }}_, rhsTime, rhsIsDuration,
+			{{- else if $AIsString }} rhsDur, rhsTime, rhsIsDuration,
+			{{- end}}  err := convertStringToDuration(b.ctx.GetSessionVars().StmtCtx, buf1.GetString(i), int8(b.tp.Decimal))
+			if err != nil  {
+				return err
+			}
+			{{- if $AIsDuration }}
+			if !rhsIsDuration {
+				result.SetNull(i, true)
+				continue
+			}
+			rhs = rhsDur
+			{{- else if $AIsTime }}
+			if rhsIsDuration {
+				result.SetNull(i, true)
+				continue
+			}
+			{{- end }}
+		{{- else if $BIsTime }} 
+			rhsTime := arg1[i]
+		{{- else }} 
+			rhs.Duration = arg1[i]
+		{{- end }}
+
+		{{- if and $AIsString $BIsString }}
+			if lhsIsDuration != rhsIsDuration {
+				result.SetNull(i, true)
+				continue
+			}
+			var (
+				d types.Duration
+				isNull bool
+			)
+			if lhsIsDuration {
+				d, isNull, err = calculateDurationTimeDiff(b.ctx, lhsDur, rhsDur)
+			} else {
+				d, isNull, err = calculateTimeDiff(b.ctx.GetSessionVars().StmtCtx, lhsTime, rhsTime)
```

Comment:
How about using a variable to store `StmtCtx` outside this loop so that we don't need call `GetSessionVars()` each time.

Reply:
I had trail this implement.

Reply:
I will try it again.